### PR TITLE
Profile App

### PR DESCRIPTION
**Pull request by casonova**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,62 @@
...
+## Installation
+To install and run the project locally, follow the steps:
+
+1. Clone the repository:
+
+
+```bash
+git clone https://github.com/casonova/Login-Application.git
+```
+
```
Comment (AminaNaqvi) [inline README.md]: Before installing dependencies, you should create virtual environment first.

File: profileapp/admin.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+
+class ProfileAdmin(admin.ModelAdmin):
+    
+    list_display = ('id', 'user', 'name','phone_number','profile_img')
+    search_fields=['phone_number']
+    list_filter=['phone_number']
+    
+
+admin.site.register(Profile,ProfileAdmin)
```
Comment (AminaNaqvi) [inline profileapp/admin.py]: 1 space here after ","

File: profileapp/models.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from django.db import models
+from django.contrib.auth.models import User
+from django.core.validators import RegexValidator
+
+
+class Profile(models.Model):
+
```
Comment (AminaNaqvi) [inline profileapp/models.py]: there is no need to add blank line here

File: profileapp/views.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+from django.shortcuts import render, redirect
+from django.contrib.auth import authenticate, login, logout
+from django.http import HttpResponse, HttpResponseNotFound, HttpResponseServerError
+from django.contrib.auth.models import User
+from django.contrib import messages
+from .forms import ProfileForm
+from django.views import View
+
+
+
```
Comment (AminaNaqvi) [inline profileapp/views.py]: before class definition only 2 blank lines are needed

File: profileapp/views.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    template_name = 'profile.html'
+    form_class = ProfileForm
+
+    def get(self, request, *args, **kwargs):
+        user_instance = request.user.profile
+        form = self.form_class(instance=user_instance)
+        return render(request, self.template_name, {'form': form})
+
+    def post(self, request, *args, **kwargs):
+        user_instance = request.user.profile
```
Comment (AminaNaqvi) [inline profileapp/views.py]: - Check if user profile exists in try except
- Check user authentication here. If user is not authenticated return reponse like "you should log in first"

File: profileapp/views.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+from django.views import View
+
+
+
+class ProfileView(View):
+    template_name = 'profile.html'
+    form_class = ProfileForm
+
+    def get(self, request, *args, **kwargs):
+        user_instance = request.user.profile
```
Comment (AminaNaqvi) [inline profileapp/views.py]: Check if user profile exists in try except

File: README.md (hunk 1 above)
Comment (casonova) [reply README.md]: I have updated the readme

File: profileapp/admin.py (hunk 2 above)
Comment (casonova) [reply profileapp/admin.py]: Add spaces between

File: profileapp/models.py (hunk 3 above)
Comment (casonova) [reply profileapp/models.py]: Removed blank line

File: profileapp/views.py (hunk 4 above)
Comment (casonova) [reply profileapp/views.py]: Remove extra line space

File: profileapp/views.py (hunk 5 above)
Comment (casonova) [reply profileapp/views.py]: Add get_object_or_404, with try-expect

File: profileapp/views.py (hunk 6 above)
Comment (casonova) [reply profileapp/views.py]: Done

**Pull request merged**